balena/models/device.py

```diff
@@ -1197,3 +1202,95 @@ def get_supervisor_state(self, uuid):
             '/supervisor/v1/device', 'POST', data=data,
             endpoint=self.settings.get('api_endpoint')
         )
+
+    def __check_os_update_target(self, device_info, target_os_version):
+        """
+        """
+
+        if 'uuid' not in device_info and not device_info['uuid']:
+            raise exceptions.OsUpdateError('The uuid of the device is not available')
+
+        if 'is_online' not in device_info and not device_info['is_online']:
+            raise exceptions.OsUpdateError('The device is offline: {uuid}'.format(uuid=uuid))
+
+        if 'os_version' not in device_info and not device_info['os_version']:
+            raise exceptions.OsUpdateError('The current os version of the device is not available: {uuid}'.format(uuid=uuid))
+
+        if 'device_type' not in device_info and not device_info['device_type']:
+            raise exceptions.OsUpdateError('The device type of the device is not available: {uuid}'.format(uuid=uuid))
+
+        if 'os_variant' not in device_info:
+            raise exceptions.OsUpdateError('The os variant of the device is not available: {uuid}'.format(uuid=uuid))
+
+        current_os_version = self.device_os.get_device_os_semver_with_variant(device_info['os_version'], device_info['os_variant'])
+        self.hup.get_hup_action_type(device_info['device_type'], current_os_version, target_os_version)
+
+    def start_os_update(self, uuid, target_os_version):
+        """
+        Start an OS update on a device.
+
+        Args:
+            uuid (str): device uuid.
+            target_os_version (str): semver-compatible version for the target device.
+                Unsupported (unpublished) version will result in rejection.
+                The version **must** be the exact version number, a "prod" variant and greater than the one running on the device.
+
+        Returns:
+            dict: action response.
+
+        Raises:
+            DeviceNotFound: if device couldn't be found.
+            InvalidParameter|OsUpdateError: if target_os_version is invalid.
+
+        Examples:
+            >>> balena.models.device.start_os_update('b6070f4fea5edf808b576123157fe5ec', '2.29.2+rev1.prod')
+            {u'status': u'in_progress', u'action': u'resinhup', u'parameters': {u'target_version': u'2.29.2+rev1.prod'}, u'last_run': 1554490809219L}
+        """
+
+        device = self.get(uuid)
+        # this will throw an error if the action is not available
+        self.__check_os_update_target(device, target_os_version)
+
+        all_versions = self.device_os.get_supported_versions(device['device_type'])['versions']
+        if not [i for i in all_versions if semver.compare(target_os_version, i) == 0]:
```

Comment:
NIT: How about `v` instead of `i` ?  :)
```suggestion
        if not [v for v in all_versions if semver.compare(target_os_version, v) == 0]:
```